home/modules/secureEnv/onePassword.nix

```diff
@@ -17,21 +17,45 @@ let
     else
       ''${pkgs.libsecret}/bin/secret-tool lookup namespace '${namespace}' key '${key}' '';
 
+  storeSshKeyCmd = keyfile: value:
+    ''echo "${value}" | ${pkgs.openssl}/bin/openssl pkcs8 -topk8 -nocrypt >${keyfile} && \
+      chmod 600 ${keyfile} && \
+      ssh-add ${keyfile} && \
+      rm -f ${keyfile}
+    '';
 
-  populateSecrets = namespace: values:
+  populateSecrets = namespace: variables: sshKeys:
     let
-      syncOne = key: item: ''
+      syncOneVariable = key: item: ''
         echo >&2 "  Setting ${key} <- ${item.vault}.${item.item}.${item.field}"
-        password=$(${pkgs._1password}/bin/op get item --session $token --vault ${item.vault} --fields '${item.field}' '${item.item}')
+        password=$(${pkgs._1password}/bin/op get item --session $token --vault '${item.vault}' --fields '${item.field}' '${item.item}')
         ${storePasswordCmd namespace key "$password" "Password from ${item.vault}.${item.item}.${item.field}"}
       '';
 
-      syncAll = items: lib.concatStringsSep "\n" (lib.mapAttrsToList syncOne items);
+      syncAllVariables = items: lib.concatStringsSep "\n" (lib.mapAttrsToList syncOneVariable items);
+
+      syncOneKey = key: item: ''
+        echo >&2 "  Adding ssh-key ${key} <- ${item.vault}.${item.item}.${item.field}"
+        sshKey=$(${pkgs._1password}/bin/op get item --session $token --vault '${item.vault}' --fields '${item.field}' '${item.item}')
```

Comment:
I think getting a value from 1Password can be extracted as a function so that we wouldn't have to repeat

Reply:
Yes, that's better.